_posts/2019-07-08-persistent-storage-improvements.md

```diff
@@ -0,0 +1,261 @@
+  kafka:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 200Gi
+      deleteClaim: false
+  zookeeper:
+    # ...
+    storage:
+      type: persistent-claim
+      size: 1000Gi
+      deleteClaim: false
+```
+
+And the rest will be taken care of by Strimzi.
+The cluster operator will automatically change the requested volume size in the PVCs and wait until the restart of the pod is required.
+Once the condition of the PVC is set to `FileSystemResizePending`, Strimzi will automatically restart the pod using this PVC.
+
+```yaml
+# ...
+status:
+  phase: Bound
+  accessModes:
+    - ReadWriteOnce
+  capacity:
+    storage: 100Gi
+  conditions:
+    - type: FileSystemResizePending
+      status: 'True'
+      lastProbeTime: null
+      lastTransitionTime: '2019-02-27T12:25:30Z'
+      message: >-
+        Waiting for user to (re-)start a pod to finish file system resize of
+        volume on node.
+```
+
+The restart might not happen immediately - it will happen as part of the next periodical reconciliation.
+So you might need to wait couple minutes before it happens.
+The cluster operator is of course aware of your whole Kafka cluster and will not restart all your pods at the same time but one by one to make sure your cluster will be still in usable state.
+
+## Adding and removing volumes from JBOD storage
+
+JBOD (Just a Bunch Of Disks) storage allows you to use multiple disks in each Kafka broker for storing commit logs.
+Strimzi added support for JBOD storage in Kafka brokers already in version 0.11 (JBOD storage is not supported in Zookeeper).
+But it didn't allowed you to add or remove volumes from it.
+You had to specify all your volumes already when creating the cluster.
+
+In 0.12 we significantly improved the JBOD storage support.
+You can of course resize the volumes as described in the previous section.
+But you can now also add or remove volumes in an existing cluster.
+
+To be able to do that, you need to already have an existing Kafka cluster using JBOD storage.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+# ...
+```
+
+If that is what you have, you can easily add more volumes just by editing the YAML.
+For example:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    # ...
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 10
+        type: persistent-claim
+        size: 1000Gi
+        deleteClaim: false
+      - id: 11
```

Comment:
why not id 2 and 3 for the new volumes?

Reply:
That is to make it more clear that the IDs don't have to be in order.